src/main/java/space/obminyashka/items_exchange/controller/AuthController.java

```diff
@@ -104,6 +112,9 @@ public ResponseEntity<String> registerUser(@RequestBody @Valid UserRegistrationD
                     ResponseMessagesHandler.ValidationMessage.USERNAME_EMAIL_DUPLICATE));
         }
 
+        EmailConfirmationToken token = new EmailConfirmationToken(null, UUID.randomUUID().toString(),
```

Comment:
Why not to edit a constructor instead of passing null?
The variable has to be as much closer to its invocation as possible.